ariadne/cardano/src/Ariadne/Wallet/Backend/KeyStorage.hs

```diff
@@ -544,6 +554,20 @@ removeSelection acidDb WalletFace{..} walletSelRef runCardanoMode = do
     fromRootId :: HdRootId -> AddressHash PublicKey
     fromRootId (HdRootId (InDb x)) = x
 
+    getAccounts :: HdWallets -> HdRootId -> [HdAccount]
+    getAccounts wallets rootId = toList $ fromRight
+      (error "Bug: UnknownHdRoot")
+      (readAccountsByRootId rootId wallets)
+
+    getAddressesByRoot :: HdWallets -> HdRootId -> [HdAddress]
+    getAddressesByRoot wallets rootId = toList $ fromRight 
+      (error "Bug: UnknownHdRoot")
+      (readAddressesByRootId rootId wallets) 
+
+    getAddressesByAccount :: HdWallets -> HdAccountId -> [HdAddress]
+    getAddressesByAccount wallets accId = toList $ fromRight
+      (error "Bug: UnknownHdAccount")
```

Comment:
These three `error`s look unpleasant and can be avoided if you move this logic into `Ariadne.Wallet.Cardano.Kernel.DB.HdWallet.Delete` (there you can delete all addresses corresponding to given `AccountId`).